rpc/types/src/misc/tx_entry.rs

```diff
@@ -37,23 +37,118 @@ pub struct TxEntry {
+                }
+            };
+        }
+
+        read_epee_field! {
+            as_hex,
+            as_json,
+            block_height,
+            block_timestamp,
+            confirmations,
+            double_spend_seen,
+            in_pool,
+            output_indices,
+            prunable_as_hex,
+            prunable_hash,
+            pruned_as_hex,
+            received_timestamp,
+            relayed,
+            tx_hash
+        }
+
+        Ok(true)
+    }
+
+    fn finish(self) -> error::Result<TxEntry> {
+        const ELSE: error::Error = error::Error::Format("Required field was not found!");
+        Ok(TxEntry {
+            as_hex: self.as_hex.ok_or(ELSE)?,
+            as_json: self.as_json.ok_or(ELSE)?,
+            block_height: self.block_height.ok_or(ELSE)?,
+            block_timestamp: self.block_timestamp.ok_or(ELSE)?,
+            confirmations: self.confirmations.ok_or(ELSE)?,
+            double_spend_seen: self.double_spend_seen.ok_or(ELSE)?,
+            in_pool: self.in_pool.ok_or(ELSE)?,
+            output_indices: self.output_indices.ok_or(ELSE)?,
+            prunable_as_hex: self.prunable_as_hex.ok_or(ELSE)?,
+            prunable_hash: self.prunable_hash.ok_or(ELSE)?,
+            pruned_as_hex: self.pruned_as_hex.ok_or(ELSE)?,
+            received_timestamp: self.received_timestamp.ok_or(ELSE)?,
+            relayed: self.relayed.ok_or(ELSE)?,
+            tx_hash: self.tx_hash.ok_or(ELSE)?,
+        })
+    }
+}
+
+#[cfg(feature = "epee")]
+impl EpeeObject for TxEntry {
+    type Builder = __TxEntryEpeeBuilder;
+
+    fn number_of_fields(&self) -> u64 {
+        // TODO: this is either 12 or 10 depending on `self.in_pool`.
+        14
+    }
+
+    fn write_fields<B: BufMut>(self, w: &mut B) -> error::Result<()> {
+        write_field(self.as_hex, "as_hex", w)?;
+        write_field(self.as_json, "as_json", w)?;
+        write_field(self.double_spend_seen, "double_spend_seen", w)?;
+        write_field(self.in_pool, "in_pool", w)?;
+        write_field(self.prunable_as_hex, "prunable_as_hex", w)?;
+        write_field(self.prunable_hash, "prunable_hash", w)?;
+        write_field(self.pruned_as_hex, "pruned_as_hex", w)?;
+        write_field(self.tx_hash, "tx_hash", w)?;
+
+        // The following section is why custom epee (de)serialization exists.
+        //
+        // <https://github.com/monero-project/monero/blob/cc73fe71162d564ffda8e549b79a350bca53c454/src/rpc/core_rpc_server_commands_defs.h#L406-L427>
+
+        if self.in_pool {
+            write_field(self.block_height, "block_height", w)?;
+            write_field(self.confirmations, "confirmations", w)?;
+            write_field(self.block_timestamp, "block_timestamp", w)?;
+            write_field(self.output_indices, "output_indices", w)?;
+        } else {
+            write_field(self.relayed, "relayed", w)?;
+            write_field(self.received_timestamp, "received_timestamp", w)?;
+        }
+
+        Ok(())
+    }
```

Comment:
@Boog900 Can you add support for a custom `EpeeObject::write_fields` in the epee macro?

All the custom epee stuff need context from `self`, so the macro's custom `write_field` doesn't work. E.g. here we need access to `self.in_pool`, then serialize fields based off that. Original code: https://github.com/monero-project/monero/blob/cc73fe71162d564ffda8e549b79a350bca53c454/src/rpc/core_rpc_server_commands_defs.h#L242-L259.

This forces us to do manual epee which gets pretty messy because of all the related boilerplate.

Reply:
I would make this type an enum of 2 structs (epee objects) and call `write_fields`/ `number_of_fields` on whatever object is being written

Reply:
Also is this type even a bin type or just JSON?

Reply:
Ah my bad, this is only JSON.